Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp

```diff
@@ -330,8 +330,11 @@ static gboolean wpeDisplayDRMSetup(WPEDisplayDRM* displayDRM, const char* device
         auto scale = parseDouble(trimmedScaleString, parsedLength);
         if (parsedLength == trimmedScaleString.length() && scaleIsInBounds(scale))
             scaleFromEnvironment = scale;
-        else
-            g_warning_once("Invalid WPE_DRM_SCALE='%*s' value, or out of bounds.", static_cast<int>(scaleString.span8().size()), scaleString.span8().data());
+        else {
+            static std::atomic<bool> warnedOnce = false;
+            if (!warnedOnce.exchange(true))
+                g_warning("Invalid WPE_DRM_SCALE='%*s' value, or out of bounds.", static_cast<int>(scaleString.span8().size()), scaleString.span8().data());
+        }
```

Comment:
I would just use `g_warning` without the atomic. It' no big deal to have a line printed each time a web view is created, because that's not an operation that would be happening the whole time, only seldomly.

```suggestion
        g_warning("Invalid WPE_DRM_SCALE='%*s' value, or out of bounds.", ```

Reply:
You don't expect people will try to use WPE to build a tabbed browser? I saw rumblings that Emacs developers were considering switching from WebKitGTK to WPE, for example. Not sure how serious that is, though.

Notably, I also notice that this build failure was already fixed in GLib several years ago.

Reply:
(But this code doesn't look like it's intended to be multithreaded, so I agree that using an atomic is overkill.)

Reply:
Yes, we expect people to use multiple web views with WPE these days, but IMO one warning for each web view that gets created is not such a big deal, just a tiny little itsy bit of log spamming. I'd rather not bring in additional code and just use `g_warning` until our GLib requirement is high enough to be able to use `g_warning_once` 😉